BASE/src/Microsoft.ApplicationInsights/Channel/Transmission.cs

```diff
@@ -209,12 +214,24 @@ public virtual async Task<HttpWebResponseWrapper> SendAsync()
                         }
                     }
                     catch (OperationCanceledException)
-                    {                        
+                    {
                         wrapper = new HttpWebResponseWrapper
                         {
                             StatusCode = (int)HttpStatusCode.RequestTimeout,
                         };
                     }
+                    finally
+                    {
+                        try
+                        {
+                            // Initiates event notification to subscriber with Transmission and TransmissionStatusEventArgs.
+                            this.TransmissionStatusEvent?.Invoke(this, new TransmissionStatusEventArgs(wrapper ?? new HttpWebResponseWrapper() { StatusCode = 0 }));
+                        }
+                        catch (Exception ex)
+                        {
+                            CoreEventSource.Log.TransmissionStatusEventError(ex.ToInvariantString());
```

Comment:
lets do the .ToInvariantString() only if listener is active. (non event in EventSource approach)

Reply:
We don't have this pattern in `CoreEventSource`, Should we introduce it?